controller/api/api.go

```diff
@@ -612,10 +612,19 @@ func (a *Api) scaleContainer(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	if err := a.manager.ScaleContainer(containerId, numInstances); err != nil {
+	result := a.manager.ScaleContainer(containerId, numInstances)
+	if len(result.Errors) > 0 {
```

Comment:
Re-ordered this so that we check `result.Errors` first.. before attempting to marshal `result`.